File: src/storage/store.ts
import { Order, Position, PortfolioSnapshot } from "../types.js";

export class InMemoryStore {
  orders: Map<string, Order> = new Map();
  positions: Map<string, Position> = new Map();
  realizedPnl = 0;
  equity = 1_000_000; // default equity INR

  getSnapshot(): PortfolioSnapshot {
    const positions = Array.from(this.positions.values());
    return {
      equity: this.equity,
      realizedPnl: this.realizedPnl,
      unrealizedPnl: 0,
      positions
    };
  }
}

File: python_app/algo_trading_py/storage/store.py
from __future__ import annotations

from dataclasses import dataclass, field

from algo_trading_py.types import Order, Position


@dataclass(slots=True)
class InMemoryStore:
  orders: dict[str, Order] = field(default_factory=dict)
  positions: dict[str, Position] = field(default_factory=dict)
  realized_pnl: float = 0.0
  equity: float = 1_000_000.0
